app/views/secrets/show.html.erb

```diff
@@ -19,21 +19,35 @@
 
   <%= form_for OpenStruct.new(secret), as: :secret, url: url, method: method, html: {class: "form-horizontal"} do |form| %>
     <fieldset>
-      <%= form.input :environment_permalink, label: "Environment" do %>
-        <%= form.select :environment_permalink,
-          options_for_select(environment_permalinks, secret[:environment_permalink]),
-          {include_blank: true}, class: "form-control", required: true, disabled: !!id %>
+      <%= form.input :environment_permalink, label: "Environment", required: true do %>
+        <% if id %>
+          <%= form.text_field :environment_permalink, disabled: true, class: "form-control" %>
+        <% else %>
+          <%= form.select :environment_permalink,
+            options_for_select(environment_permalinks, secret[:environment_permalink]),
+            {include_blank: true}, required: true, **Samson::FormBuilder::LIVE_SELECT_OPTIONS %>
+        <% end %>
       <% end %>
 
-      <%= form.input :project_permalink, label: "Project" do %>
-        <%= live_select_tag 'secret[project_permalink]',
-          options_for_select(@writable_project_permalinks, secret[:project_permalink]), include_blank: true, required: true, disabled: !!id %>
+      <%= form.input :project_permalink, label: "Project", required: true do %>
+        <% if id %>
+          <%= form.text_field :project_permalink, disabled: true, class: "form-control" %>
+        <% else %>
+          <%= form.select :project_permalink,
+            options_for_select(@writable_project_permalinks, secret[:project_permalink]),
+            {include_blank: true}, required: true, **Samson::FormBuilder::LIVE_SELECT_OPTIONS
+          %>
+        <% end %>
       <% end %>
 
-      <%= form.input :deploy_group_permalink, label: "Deploy Group" do %>
-        <%= form.select :deploy_group_permalink,
-          options_for_select([['Loading ...', secret[:deploy_group_permalink]]], secret[:deploy_group_permalink]),
-          {include_blank: true}, class: "form-control", required: true, disabled: !!id %>
+      <%= form.input :deploy_group_permalink, label: "Deploy Group", required: true do %>
+        <% if id %>
+          <%= form.text_field :deploy_group_permalink, disabled: true, class: "form-control" %>
+        <% else %>
+          <%= form.select :deploy_group_permalink,
+            options_for_select([['Loading ...', secret[:deploy_group_permalink]]], secret[:deploy_group_permalink]),
+            {include_blank: true}, class: "form-control", required: true %>
```

Comment:
the only one that cannot be a live select since we dynamically change the select values